lxd/images.go

```diff
@@ -3559,20 +3559,34 @@ func imageAliasesGet(d *Daemon, r *http.Request) response.Response {
 			responseStr = make([]string, 0, len(names))
 		}
 
+		visited := make(map[string]map[int]bool)
+
 		for _, name := range names {
-			if !userHasPermission(entity.ImageAliasURL(projectName, name)) {
-				continue
+			if visited[name] == nil {
+				visited[name] = make(map[int]bool)
 			}
 
-			if !recursion {
-				responseStr = append(responseStr, api.NewURL().Path(version.APIVersion, "images", "aliases", name).String())
-			} else {
-				_, alias, err := tx.GetImageAlias(ctx, projectName, name, "", true)
-				if err != nil {
+			for _, imageType := range imageTypes {
+				if visited[name][imageType] {
+					continue
+				}
+
+				if !userHasPermission(entity.ImageAliasURL(projectName, name)) {
```

Comment:
@markylaing Now we can have two aliases with the same name (one for a container image and one for a VM image), but both would have the same URL and this I understand that the same permissions would be granted for both.
I think maybe this is best kept as is and we just grant the same permissions for aliases with the same name, but I would like to double check and ask how do you think we should handle this.

Reply:
we cant have 2 different resources with the same URL - this might block the change indeed

Reply:
so whilst i am not too concerned about the authz part of this, we need to check if the image alias itself is addressable over the API, and if we have to introduce a breaking API change for this then we might have to can the idea.